### PR TITLE
OEUI-84: Integrate generic loader for careSettings, encounterType, orderEntryActions and settingEncounterType

### PR DESCRIPTION
**Pull request by fuchodeveloper**
## JIRA TICKET NAME

[OEUI-84:  Integrate generic loader for careSettings, encounterType, orderEntryActions and settingEncounterType](https://issues.openmrs.org/browse/OEUI-84)

## SUMMARY:
This ticket implements integration of generic loader for careSetting, encounterType, orderEntryActions and settingEncounterType actions.

### CODE REVIEW

File: tests/reducers/orderEntryReducer.test.js [hunk 1]
```diff
@@ -37,15 +37,7 @@ describe('Order Entry Configurations reducer', () => {
         const actualState = orderEntryConfigurations(initialState, fetchConfigurationsSuccess(configurations));
         expect(actualState).toEqual(expectedState);
     });
-    it('should handle `FETCH_ORDER_CONFIG_LOADING`', () => {
```
Comment (Betty-Kebenei) [inline tests/reducers/orderEntryReducer.test.js]: Why delete this?

File: tests/reducers/settingEncounterTypeReducer.test.js [hunk 2]
```diff
@@ -36,17 +36,7 @@ describe('Setting Encounter Type reducer', () => {
         );
         expect(actualState).toEqual(expectedState);
     });
-    it('should handle `SETTING_ENCOUNTER_TYPE_LOADING`', () => {
```
Comment (Betty-Kebenei) [inline tests/reducers/settingEncounterTypeReducer.test.js]: Why delete this too?

File: tests/reducers/orderEntryReducer.test.js (hunk 1 above)
Comment (fuchodeveloper) [reply tests/reducers/orderEntryReducer.test.js]: The deleted `it` block calls `fetchConfigurationsLoading` in `orderEntryActions.js` action which is no longer used in the app.

File: tests/reducers/settingEncounterTypeReducer.test.js (hunk 2 above)
Comment (fuchodeveloper) [reply tests/reducers/settingEncounterTypeReducer.test.js]: Similarly, the `it` block calls `settingEncounterTypeLoading` in `settingEncounterType.js` action which is no longer used in the app.

**Comment by flacode** [comment]
You did not delete the loading actions from actionTypes.js

File: app/js/actions/settingEncounterType.js [hunk 3]
```diff
@@ -2,9 +2,9 @@ import axios from 'axios';
 import {
   SETTING_ENCOUNTER_TYPE_SUCCESS,
   SETTING_ENCOUNTER_TYPE_FAILURE,
-  SETTING_ENCOUNTER_TYPE_LOADING,
```
Comment (flacode) [inline app/js/actions/settingEncounterType.js]: Removing this line would also require removing the actionType from the test file.

Comment (fuchodeveloper) [reply app/js/actions/settingEncounterType.js]: Thanks for your feedback. 
Based on the implementation test by @Betty-Kebenei, the loading action type is needed by other test blocks in the `settingEncounterType.test.js` file. 
This PR takes care of the `fetchConfigurationsLoading` created in `settingEncounter.js` file which was the reason for removing it.

File: app/js/actions/careSetting.js [hunk 4]
```diff
@@ -16,24 +16,19 @@ const fetchPatientCareSettingError = error => ({
...
-      dispatch(fetchPatientCareSettingLoading(false));
+      dispatch(loading('PATIENT_CARESETTING', false));
     })
     .catch((error) => {
       if (!error.response) {
         dispatch(networkError('Network error occurred'));
       } else {
         dispatch(fetchPatientCareSettingError(error.response));
-        dispatch(fetchPatientCareSettingLoading(false));
+        dispatch(loading('PATIENT_CARESETTING', false));
```
[the remark this answers is not in the document]
Comment (fuchodeveloper) [reply app/js/actions/careSetting.js]: @larrystone thank you for your suggestion.
I have updated the PR to reflect the changes

[the remark this answers is not in the document]
Comment (larrystone) [reply app/js/actions/careSetting.js]: Okay, nice work

**Pull request merged**